Add Typesafe `<NavLink>` component to Wasp (#4104)
wasp-lang/wasp: web/docs/advanced/links.md

}
```

> The `to` prop is autocompleted from the routes you defined in `main.wasp`, and `params` is typechecked against the path you picked. Rename a route or change a param, and any broken `Link` lights up at compile time.

Reviewer: ```suggestion
The `to` prop is autocompleted from the routes you defined in `main.wasp`, and `params` is typechecked against the path you picked. Rename a route or change a param, and any broken `Link` is pointed out by Typescript.
```

- all other props that the `react-router`'s [Link](https://reactrouter.com/7.12.0/api/components/Link) component accepts

> ### `NavLink` Component

Reviewer: Nitpick: If there is a way for us to maybe DRY up the API reference between `Link` and `NavLink` that would be cool.

Author: I think for API references that is okay.
If we ever automate this (e.g. from JSDocs), it would end up the same as it is now.
So as not to change and then regress, I think this is okay.
